### PR TITLE
[kf5-kcalendarcore] Fix last modified time for new exception. JB#55657

### PR DESCRIPTION
**Pull request by bshmarin**

### CODE REVIEW

**Comment by dcaliste** [comment]
No problem with the commit, but could you provide an explanation on what is failing and what bug this modification is addressing ? Thanks in advance.

File: rpm/0003-Fix-last-modified-time-for-new-exception.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
++  SPDX-FileCopyrightText: 2021 Boris Shmarin <b.shmarin@omp.ru>
+ 
+   SPDX-License-Identifier: LGPL-2.0-or-later
+ */
+@@ -584,7 +585,9 @@ Incidence::Ptr Calendar::createException(const Incidence::Ptr &incidence, const
+     }
+ 
+     Incidence::Ptr newInc(incidence->clone());
+-    newInc->setCreated(QDateTime::currentDateTimeUtc());
++    QDateTime current = QDateTime::currentDateTimeUtc();
```
Comment (dcaliste) [inline rpm/0003-Fix-last-modified-time-for-new-exception.patch]: Not sure it's mandatory but you may declare this `current` variable `const`.

**Comment by bshmarin** [comment]
> No problem with the commit, but could you provide an explanation on what is failing and what bug this modification is addressing ? Thanks in advance.

Expanded bug description: https://bz.jollamobile.com/show_bug.cgi?id=55657

File: rpm/0003-Fix-last-modified-time-for-new-exception.patch (hunk 1 above)
Comment (bshmarin) [reply rpm/0003-Fix-last-modified-time-for-new-exception.patch]: Done.

**Comment by dcaliste** [comment]
Thank you for the update. But for the bug description, I'm afraid this bugtracker is not public. General readers from Github and I don't have access to it.

**Comment by bshmarin** [comment]
> Thank you for the update. But for the bug description, I'm afraid this bugtracker is not public. General readers from Github and I don't have access to it.

When exception is created (in kf5-calendarcore) it consists of a copy of the recurrent event data, including the last modification date that is not updated later, and sailfish-eas doesn't synchronize this exception with the server because the last modification date is too old.

**Comment by dcaliste** [comment]
I see, in EAS, you're not using the (deprecated) mKCal::ExtendedCalendar::dissociateSingleOccurrence() anymore, right ?

Then, I agree it's an issue and I'll propose the patch upstream.

**Comment by bshmarin** [comment]
> I see, in EAS, you're not using the (deprecated) mKCal::ExtendedCalendar::dissociateSingleOccurrence() anymore, right ?

Yes, it used in the nemo-qml-plugin-calendar.

**Comment by dcaliste** [comment]
Thank you for the patch and the explanation. Here is the MR upstream with your commit : https://invent.kde.org/frameworks/kcalendarcore/-/merge_requests/58

**Comment by dcaliste** [comment]
Allan Winter accepted the patch upstream some minutes ago.

As far as I'm concerned, I'm ok with accepting the PR as is, or by upgrading to latest upstream.

**Comment by pvuorela** [comment]
With the patch upstream, i might now prefer just bumping the checkout. Could be fine even just to the latest git, but there are new versions coming quite often, wouldn't expect taking too much time.

**Comment by bshmarin** [comment]
Then maybe a patch for now?
For prevent problem with other patches...

**Comment by pvuorela** [comment]
What problems are you referring to? We've been following upstream quite well lately here and the delta is only a few commits. That's unlikely to cause conflicts. With patch introduced it's small extra work on the next update to get rid of it.

**Comment by bshmarin** [comment]
> What problems are you referring to?

if bump then:
mb2 build -p
Reversed (or previously applied) patch detected!  Assume -R? [n]

**Comment by bshmarin** [comment]
Then another PR:
https://github.com/sailfishos/kf5-calendarcore/pull/6

**Pull request closed without merging**